modules/apps/export-import/export-import-changeset-service/src/main/java/com/liferay/exportimport/changeset/internal/manager/ChangesetManagerImpl.java

```diff
@@ -63,28 +68,38 @@ public void addChangeset(Changeset changeset) {
 		_changesets.put(changesetUuid, changeset);
 	}
 
+	@Clusterable(onMaster = true)
 	@Override
 	public void clearChangesets() {
 		_changesets = new HashMap<>();
 	}
 
+	@Override
+	public String getOSGiServiceIdentifier() {
+		return ChangesetManager.class.getName();
+	}
+
+	@Clusterable(onMaster = true)
 	@Override
 	public boolean hasChangeset(String changesetUuid) {
 		return _changesets.containsKey(changesetUuid);
 	}
 
+	@Clusterable(onMaster = true)
 	@Override
 	public Optional<Changeset> peekChangeset(String changesetUuid) {
```

Comment:
@Preston-Crary this is not going to work, Optional is not Serializable, ClusterExecutor will reject it on calling.
If you want to keep this cluster solution, those non-serializable return type methods must be cleaned, but I guess that is not allowed either, since you said this is meant for backport. Wiping them off will cause major version bump.
You will probably need to look for a different solution.

Reply:
Hi @shuyangzhou , Only 2 public methods of ChangesetManagerImpl are actually used: addChangeset() and popChangeset(). Only addChangeset() should be remotely callable, since popChangeset() is always called from a background task on the master node. Those methods will work correctly. However, the ChangesetManagerImpl having only one method running on master can be quite confusing.

Reply:
@vendeltoreki add a new method to to ChangeSetManager:

`public boolean publishChangeSet(String changesetUuid, PortletDataContext portletDataContext);`

Then replace the usage of `popChangeset(String changesetUuid)` with the new method.  Use the boolean return where `isPresent()` is used.  Deprecate all the unused methods and do not annotate them.